types/use-global-hook/index.d.ts

```diff
@@ -4,23 +4,36 @@
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 // TypeScript Version: 3.6
 
-import React = require('react');
-type ReactType = typeof React;
+// Use an interface so that different versions of React can be used
+interface ReactInterface {
+  // tslint:disable-next-line ban-types
+  useEffect: Function;
```

Comment:
Just use `(...args: any[]) => any` rather than the `Function` global and a lint suppression.

Reply:
Makes sense. 👍